src/cascadia/TerminalSettingsModel/WslDistroGenerator.cpp

```diff
@@ -136,3 +143,183 @@ std::vector<Profile> WslDistroGenerator::GenerateProfiles()
+// - Skips distros that are utility distros for docker (see GH#3556)
+// Arguments:
+// - names: a list of distro names to turn into profiles
+// Return Value:
+// - the list of profiles we've generated.
+static std::vector<Profile> namesToProfiles(const std::vector<std::wstring>& names)
+{
+    std::vector<Profile> profiles;
+    for (const auto& distName : names)
+    {
+        if (distName.substr(0, std::min(distName.size(), DockerDistributionPrefix.size())) == DockerDistributionPrefix)
+        {
+            // Docker for Windows creates some utility distributions to handle Docker commands.
+            // Pursuant to GH#3556, because they are _not_ user-facing we want to hide them.
+            continue;
+        }
+
+        auto WSLDistro{ CreateDefaultProfile(distName) };
+
+        WSLDistro.Commandline(L"wsl.exe -d " + distName);
+        WSLDistro.DefaultAppearance().ColorSchemeName(L"Campbell");
+        WSLDistro.StartingDirectory(DEFAULT_STARTING_DIRECTORY);
+        WSLDistro.Icon(L"ms-appx:///ProfileIcons/{9acb9455-ca41-5af7-950f-6bca1bc9722f}.png");
+        profiles.emplace_back(WSLDistro);
+    }
+    return profiles;
+}
+
+// Function Description:
+// - Open the reg key the root of the WSL data, in HKCU\Software\Microsoft\Windows\CurrentVersion\Lxss
+// Arguments:
+// - <none>
+// Return Value:
+// - the HKEY if it exists and we can read it, else nullptr
+static HKEY openWslRegKey()
+{
+    HKEY hKey{ nullptr };
+    if (RegOpenKeyEx(HKEY_CURRENT_USER, RegKeyLxss, 0, KEY_ALL_ACCESS, &hKey) == ERROR_SUCCESS)
+    {
+        return hKey;
+    }
+    return nullptr;
+}
+
+// Function Description:
+// - Open the reg key for a single distro, underneath the root WSL key.
+// Arguments:
+// - wslRootKey: the HKEY for the Lxss node.
+// - guid: the string representation of the GUID for the distro to inspect
+// Return Value:
+// - the HKEY if it exists and we can read it, else nullptr
+static HKEY openDistroKey(const wil::unique_hkey& wslRootKey, const std::wstring& guid)
+{
+    HKEY hKey{ nullptr };
+    if (RegOpenKeyEx(wslRootKey.get(), guid.c_str(), 0, KEY_ALL_ACCESS, &hKey) == ERROR_SUCCESS)
+    {
+        return hKey;
+    }
+    return nullptr;
+}
+
+// Function Description:
+// - Get the list of all the guids of all the WSL distros from the registry. If
+//   we fail to open or read the root reg key, we'll return false.
+//   Places the guids of all the distros into the "guidStrings" param.
+// Arguments:
+// - wslRootKey: the HKEY for the Lxss node.
+// - names: a vector that receives all the guids of the installed distros.
+// Return Value:
+// - false if we failed to enumerate all the WSL distros
+static bool getWslGuids(const wil::unique_hkey& wslRootKey,
+                        std::vector<std::wstring>& guidStrings)
+{
+    if (!wslRootKey)
+    {
+        return false;
+    }
+
+    wchar_t buffer[39]; // a {GUID} is 38 chars long
+    for (DWORD i = 0;; i++)
+    {
+        DWORD length;

```

Comment:
BTW I had a programming error. This value needs to be initialized to `std::size(buffer)` (or 39).
This code shouldn't work without it. 🤔